nameable.rb

```diff
@@ -0,0 +1,32 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    return @nameable.correct_name[0..9] if @nameable.correct_name.length > 10
+
+    @nameable.correct_name
+  end
+end
```

Comment:
- According to the project requirements, each of the individual classes should be in a different file not in the same file as the `Nameable` class. Please fix this 🙏

Reply:
Oh, OK, I was missed because, in the lesson documentation, the Decorators and the "main" class are in the same file, I will change it right now! thanks for the advice.